code/modules/client/verbs/looc.dm

```diff
@@ -97,6 +118,116 @@ AUTH_CLIENT_VERB(looc, msg as text)
+		try_criticise(usr.client)
+
+/datum/looc_message/proc/try_commend(client/listener)
+	if (listener.player_details.muted & MUTE_OOC)
+		return
+	if (!(listener.ckey in hearers))
+		return
+	if (LAZYFIND(commenders, listener.ckey))
+		to_chat(listener, span_good("You have already rated this message, thank you for creating a positive atmosphere!"))
+		return
+	var/client/sender_client = sender.find_client()
+	if (sender_client)
+		if (sender.commendations_received > 10)
+			to_chat(listener, span_good("You sent a commendation to [mob_name], thank you for creating a positive atmosphere!"))
+			to_chat(sender_client, span_good("You received a commendation for being helpful, but have been so helpful that you already hit the limit! Thank you for creating a positive atmosphere!"))
+		else
+			if (CONFIG_GET(flag/grant_metacurrency))
+				listener.inc_metabalance(1, TRUE, reason = "You sent a commendation to [mob_name], thank you for creating a positive atmosphere!")
+				sender_client.inc_metabalance(1, TRUE, reason = "You have received a commendation for helpful messages, thank you for creating a positive atmosphere!")
+			else
+				to_chat(listener, span_good("You sent a commendation to [mob_name], thank you for creating a positive atmosphere!"))
+				to_chat(sender_client, span_good("You have received a commendation for helpful messages, thank you for creating a positive atmosphere!"))
+	LAZYADD(commenders, listener.ckey)
+	// Maximum of 20 commendations, which protects you from 3 criticisms, to discourage building
+	// a clique and talking in LOOC for commendations
+	if (sender.commendations_received < 20)
+		sender.commendations_received ++
+
+/datum/looc_message/proc/try_criticise(client/listener)
+	if (listener.player_details.muted & MUTE_OOC)
+		return
+	if (tgui_alert(listener, "Are you sure you want to criticise this message?", "Downvote message", list("Criticize", "Abort")) == "Abort")
+		return
+	if (!(listener.ckey in hearers))
+		return
+	if (listener.player_details.has_criticized)
+		to_chat(listener, span_warning("You may only use the criticize function once per round."))
+		return
+	if (LAZYFIND(commenders, listener.ckey))
+		to_chat(listener, span_good("You have already rated this message, thank you for maintaining mutual respect even when the game gets tough."))
+		return
+	to_chat(listener, span_good("You criticised a message from [mob_name], thank you for maintaining mutual respect even when the game gets tough."))
+	LAZYADD(commenders, listener.ckey)
+	var/client/sender_client = sender.find_client()
+	listener.player_details.has_criticized = TRUE
+	// Do nothing instead
+	if (sender_client && sender_client.holder)
+		return
+	sender.criticisms_received ++
+	// Will hold a reference until complete, at which point the datum will be deleted
+	if (!timer_active)
+		addtimer(CALLBACK(src, PROC_REF(issue_warning)), rand(2 MINUTES, 5 MINUTES))
+		timer_active = TRUE
+
+/datum/looc_message/proc/issue_warning()
```

Comment:
Log successful mutes

Reply:
I have logged successful muted. I have not logged critique or commendation actions since I believe that this system works best when all users feel free to commend/critique a message for whatever reason.